backend/curation/api/v1/curation/collections/collection_id/datasets/dataset_id/actions.py

```diff
@@ -87,16 +88,20 @@ def delete(token_info: dict, collection_id: str, dataset_id: str, delete_publish
     if dataset_version.version_id not in [v.version_id for v in collection_version.datasets]:
         raise ForbiddenHTTPException(f"Dataset {dataset_id} does not belong to a collection")
 
-    delete_published_flag = False
     if user_info.is_cxg_admin() and delete_published == "true" and not collection_version.published_at:
-        delete_published_flag = True
-
-    try:
-        business_logic.remove_dataset_version(
-            collection_version.version_id, dataset_version.version_id, delete_published_flag
-        )
-    except CollectionUpdateException:
-        raise MethodNotAllowedException(detail="Cannot delete a public Dataset") from None
+        try:
+            business_logic.remove_dataset_version(
+                collection_version.version_id, dataset_version.version_id, delete_published=True
+            )
+        except DatasetIsNotPublishedException:
+            raise InvalidParametersHTTPException(
```

Comment:
My initial thinking was if someone tries to perform an action that indicates they have a misunderstanding of the state of the data, we want them to know that, rather than returning 2xx. They might've been trying to delete the wrong Dataset.

Sort of similar to how we force users to use the appropriate `/datasets/` vs `/dataset_versions/` route for retrieving Datasets even though we could have just used fall-through under a single route.

But I could be convinced to not fail here. Perhaps @brianraymor would like to weigh in.

Reply:
for deletion, i'm happy to play it more conservative.

Reply:
great 🤝